### DISCUSSION TITLE
how to get the content type value of a specific version of an item?

### DISCUSSION THREAD

**Discussion by jagsridharan** [pnp/pnpcore#1442] [Q&A]
Hi,

I am trying to capture the metadata for Documents for the current version and as well all the previous version. I tried using Graph API and I am able to get all the Field values for the previous versions. However, I was not able to get the "ContentType" property of previous versions of the document. I thought this is not yet. exposed in Graph and tried the CSOM, still no luck. I can get a lot of properties under "Fields" including the actual column values, however there is no way I could get the "ContentType" used on a given version.

I cannot use the current version's Content Type, as this may not be the same in the previous versions. What is the best way to get the ContentType name (or ID) of a document's previous versions.

Regards,
Jag

**Comment by jansenbe** [comment]
@jagsridharan : tried this in PnP Core SDK (which uses SharePoint REST for version retrieval) and see the same, no content type information in the version record.

**Reply by jagsridharan** [reply]
Hi @jansenbe , that's correct. I tried graph, and pnp separately and same effect. In fact, the csom method gives me the content type but it gives me the current version's conten type and not the content type used for the specific version.

**Reply by jagsridharan** [reply]
Hi @jansenbe  should this be reported as a bug to PnP Core or is it with SharePoint REST team?

**Reply by jansenbe** [reply]
Given none of the API return this one can say this is a "feature request", these need to be reported via the regular support channels. Don't think this is an issue for PnP Core SDK

**Reply by jagsridharan** [reply]
Thanks @jansenbe. I will do that. It's surprising none of the 3rd party vendors who perform backups reported this ;) being an issue.
